powerauth-java-crypto/src/main/java/io/getlime/security/powerauth/crypto/lib/encryptor/ecies/EciesRequestResponseValidator.java

```diff
@@ -53,8 +53,8 @@ public EciesRequestResponseValidator(String protocolVersion) throws EncryptorExc
         if (!supportedVersions.contains(protocolVersion)) {
             throw new EncryptorException("Unsupported protocol version " + protocolVersion);
         }
-        this.useTimestamp = "3.2".equals(protocolVersion);
-        this.useNonceForRequest = "3.2".equals(protocolVersion) || "3.1".equals(protocolVersion);
+        this.useTimestamp = "3.3".equals(protocolVersion) || "3.2".equals(protocolVersion);
```

Comment:
I think a good approach for this would be to introduce a new interface, for example `ProtocolFeatures` that will contain getters for all optional features. For example:

```java
interface ProtocolFeatures {
    boolean useTimestampForEcies();
    boolean useNonceForEciesRequest();
    boolean useTemporaryKeysForEcies();
   // etc...
}
```
Then some factory class may instantiate an object implementing this, depending on the protocol version. I would do this as a refactoring before we start with crypto 4.0 :)

Reply:
I totally agree, a feature interface is much better than the collection.

Reply:
@banterCZ Already tracked as https://github.com/wultra/powerauth-crypto/issues/528.